src/core/toga/widgets/optioncontainer.py

```diff
@@ -3,6 +3,63 @@
 from .base import Widget
 
 
+class OptionItem:
+    def __init__(self, container, label, widget, enabled):
+        self._container = container
+        self._index = len(container)
+        self._label = label
+        self._widget = widget
+        self._enabled = enabled
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        self._enabled = value
+        self._container.interface._impl.set_option_enabled(self._index, value)
```

Comment:
I'm not sure if this is a correct use of toga or I'm breaking some toga design principle

Reply:
The general idea of "invoke a method on the impl" is exactly the right idea. 

The only criticism I'd have would be around how you get to the "_impl" in the first place.